src/app-layer-smtp.c

```diff
@@ -421,131 +498,52 @@ int SMTPProcessDataChunk(const uint8_t *chunk, uint32_t len,
 
     /* Find file */
     if (entity->ctnt_flags & CTNT_IS_ATTACHMENT) {
-
-        /* Make sure file container allocated */
-        if (smtp_state->files_ts == NULL) {
-            smtp_state->files_ts = FileContainerAlloc();
-            if (smtp_state->files_ts == NULL) {
-                ret = MIME_DEC_ERR_MEM;
-                SCLogError(SC_ERR_MEM_ALLOC, "Could not create file container");
-                SCReturnInt(ret);
-            }
+        if (SMTPFileContainerAlloc(smtp_state) == MIME_DEC_ERR_MEM) {
+            SCReturnInt(-2);
```

Comment:
this is a bit odd. We use `MIME_DEC_ERR_MEM` in our util func, but do not use `MIME_DEC_...` to return from a mime api callback?